### PR TITLE
Combine workspace into a unicrate

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Andy Lok <andylokandy@hotmail.com>

### CODE REVIEW

**Comment by pingyu** [comment]
Why we change `slog` back to `log` (see https://github.com/tikv/client-rust/pull/298) ?

It will break the compatibility as we change the interface.

**Comment by Xuanwo** [comment]
> It will break the compatibility as we change the interface.

Since we have already switched to the `tonic` grpc framework, our upcoming release must be `v0.3`, which is considered a breaking change. Therefore, compatibility is not a concern and all we need to consider is whether the answer is `YES` or `NO`.

In my opinion, a library should always use `log` (maybe `tracing`) instead of a specified logger like `slog`. This makes it easier to integrate the library. And please note that `slog` itself is inactive and recommand users to migrate to `tracing` instead.

image

File: src/common/security.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2018 TiKV Project Authors. Licensed under Apache-2.0.
 
-use crate::Result;
+use crate::{internal_err, Result};
+use log::info;
 // use grpcio::{Channel, ChannelBuilder, ChannelCredentialsBuilder, Environment};
```
Comment (Xuanwo) [inline src/common/security.rs:5]: How about removing this line.

**Review by Xuanwo — COMMENTED** [review]
Mostly LGTM, thanks!

**Review by Xuanwo — APPROVED** [review]

**Review by pingyu — APPROVED** [review]
LGTM~

**Pull request merged**